ion/binarywriter.go

```diff
@@ -492,9 +519,17 @@ func (w *binaryWriter) beginValue(api string) error {
 			return &UsageError{api, "field name not set"}
 		}
 
-		id, err := w.resolve(api, *name)
-		if err != nil {
-			return err
+		var id uint64
+		if name.LocalSID != SymbolIDUnknown {
+			id = uint64(name.LocalSID)
+		} else if name.Text != nil {
+			var err error
+			id, err = w.resolve(api, *name.Text)
+			if err != nil {
+				return err
+			}
+		} else {
+			return &UsageError{api, "invalid field name symbol token"}
```

Comment:
The error message should communicate why the symbol token is invalid.

Reply:
Done.